collectors/python.d.plugin/memcachier/memcachier.chart.py

```diff
@@ -0,0 +1,204 @@
+        'lines': [
+            ['cas_hits', 'hits', 'percent-of-absolute-row'],
+            ['cas_misses', 'misses', 'percent-of-absolute-row'],
+            ['cas_badval', 'bad value', 'percent-of-absolute-row']
+        ]
+    },
+    'increment': {
+        'options': [None, 'Requests', 'requests', 'increment ops', 'memcachier.increment', 'stacked'],
+        'lines': [
+            ['incr_hits', 'hits', 'percent-of-absolute-row'],
+            ['incr_misses', 'misses', 'percent-of-absolute-row']
+        ]
+    },
+    'decrement': {
+        'options': [None, 'Requests', 'requests', 'decrement ops', 'memcachier.decrement', 'stacked'],
+        'lines': [
+            ['decr_hits', 'hits', 'percent-of-absolute-row'],
+            ['decr_misses', 'misses', 'percent-of-absolute-row']
+        ]
+    },
+    'touch': {
+        'options': [None, 'Requests', 'requests', 'touch ops', 'memcachier.touch', 'stacked'],
+        'lines': [
+            ['touch_hits', 'hits', 'percent-of-absolute-row'],
+            ['touch_misses', 'misses', 'percent-of-absolute-row']
+        ]
+    },
+    'touch_rate': {
+        'options': [None, 'Rate', 'requests/s', 'touch ops', 'memcachier.touch_rate', 'line'],
+        'lines': [
+            ['cmd_touch', 'rate', 'incremental']
+        ]
+    }
+}
+
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.url = 'https://analytics.memcachier.com/api/v1/login'
+        self.user = self.configuration.get('user')
+        self.password = self.configuration.get('pass')
+
+    def _get_data(self):
+        """
+        Get data from API
+        :return: dict
+        """
+        response = self._get_raw_data()
+        if response is None:
+            return None
+
+        metrics = json.loads(response)
+
+        # check if we have more than one server in the results
+        if len(metrics) > 1:
+            self.error('This plugin only works with one cache_server per job.')
+            self.debug('Request result: {r}'.format(r=metrics))
+            return None
+
+        data = list(metrics.values())[0]
+
+        # custom calculations
+        try:
+            data['avail'] = int(data['limit_maxbytes']) - int(data['bytes'])
+            data['used'] = int(data['bytes'])
+        except (KeyError, ValueError, TypeError):
+            pass
+
+        return data
+
+    def check(self):
+        """
+        Create manager, get cache_id and update url
+        :return: boolean
+        """
+        self._manager = self._build_manager()
+        data = self._get_raw_data()
+
+        if data is None:
+            return False
+
+        # we need to guarantee that only one cache_id is passed forward
+        data = json.loads(data)
+        if len(data) == 1 and type(data['cache_id']) == int:
```

Comment:
I am still not clear. Could we have several ids in the response or not? It is not clear from their docs.

How about to ask on [github](https://github.com/memcachier/)?